app/src/main/java/tttTDD/Java/Board.java

```diff
@@ -1,44 +1,90 @@
 package tttTDD.Java;
 
-import org.checkerframework.common.value.qual.IntRange;
-
-import java.lang.reflect.Array;
+import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 public class Board {
+    private final int DEFAULT_SIZE = 3;
     private final String[] board;
+    private int numberOfSpaces;
+    private int[][] winningSets = {
+            {0, 1, 2},
+            {3, 4, 5},
+            {6, 7, 8},
+            {0, 3, 6},
+            {1, 4, 7},
+            {2, 5, 8},
+            {0, 4, 8},
+            {2, 4, 6},
+    };
 
     public Board() {
-        this.board = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
+        this.board = initializeBoard(DEFAULT_SIZE);
     }
 
     public Board(int boardSize) {
-        this.board = new String[boardSize * boardSize];
-        for (int i = 0; i < this.board.length; i++) {
-            this.board[i] = String.valueOf(i + 1);
-        }
+        this.board = initializeBoard(boardSize);
     }
 
-    public String[] getBoard() {
-        return this.board;
+    public String getSpace(int location) {
+        return isLocationValid(location) ? null : this.board[location];
     }
 
-    public String getSpace(int location) {
-        return this.board[location];
+    public int getFirstAvailableMove() {
+        String[] availableMoves = getAvailableMoves(board);
+        return convertToLocation(availableMoves[0]);
     }
 
-    public void updateBoard(int location, String marker) throws Exception {
+    public boolean updateBoard(int location, String marker) {
         if (isSpaceAvailable(location)) {
             this.board[location] = marker;
-        } else {
-            throw new Exception("The space is already occupied");
+            return true;
+        }
+
+        return false;
```

Comment:
As we discussed in IPM, the downside to merely returning `false` (instead of throwing) when given an illegal instruction is that it's pretty easy to consume this method in such a manner that you don't realize that it didn't do anything. The current implementation of `performMove` is a good example of this.

Reply:
When it comes to handling exceptions, would you say is best to have it happen at a higher level in the app or here within the `Board` class?

Reply:
It may be appropriate to handle (catch) the exception that I'm suggesting you throw here in other methods that call `updateBoard` within the Board class, but it should _not_ be caught in the same method that throws it, because then it's just an obfuscated if statement.

Furthermore, some methods that call another function that can throw an exception have no need to handle them. If there's nothing coherent to do in response to the failure of (e.g.) the move to be made, then it may be appropriate to let the error continue to propagate up the stack.